src/__experimental__/components/radio-button/radio-button.spec.js

```diff
@@ -1,88 +1,81 @@
 import React from 'react';
-import TestRenderer from 'react-test-renderer';
+import { act } from 'react-dom/test-utils';
 import { mount } from 'enzyme';
-import 'jest-styled-components';
-import { css } from 'styled-components';
-import { RadioButton, OriginalRadioButton } from '.';
+import { css, ThemeProvider } from 'styled-components';
+import TestRenderer from 'react-test-renderer';
+import { RadioButton, RadioButtonGroup } from '.';
 import FieldHelpStyle from '../field-help/field-help.style';
 import LabelStyle from '../label/label.style';
-import CheckableInput from '../checkable-input/checkable-input.component';
 import HiddenCheckableInputStyle from '../checkable-input/hidden-checkable-input.style';
 import { StyledCheckableInput } from '../checkable-input/checkable-input.style';
 import StyledCheckableInputSvgWrapper from '../checkable-input/checkable-input-svg-wrapper.style';
 import { assertStyleMatch } from '../../../__spec_helper__/test-utils';
 import guid from '../../../utils/helpers/guid';
 import baseTheme from '../../../style/themes/base';
-import classicTheme from '../../../style/themes/classic';
+import classic from '../../../style/themes/classic';
+import { getValidationType } from '../../../components/validations/with-validation.hoc';
+import small from '../../../style/themes/small';
 
 jest.mock('../../../utils/helpers/guid');
 guid.mockImplementation(() => 'guid-12345');
 
-function render(props, renderer = TestRenderer.create) {
+function render(props = {}, theme = small, renderer = mount) {
+  const {
+    hasError, hasInfo, hasWarning, ...buttonProps
+  } = props;
+  const groupProps = {
+    hasError, hasInfo, hasWarning
+  };
   return renderer(
-    <RadioButton
-      name='my-radio'
-      value='test'
-      { ...props }
-    />
+    <ThemeProvider theme={ theme }>
+      <RadioButtonGroup { ...groupProps }>
+        <RadioButton
+          name='my-radio'
+          value='test'
+          { ...buttonProps }
+        />
+      </RadioButtonGroup>
+    </ThemeProvider>
   );
 }
 
-function renderClassic(props) {
-  return TestRenderer.create(
-    <RadioButton
-      name='my-radio'
-      theme={ classicTheme } value='test'
-      { ...props }
-    />
-  );
-}
+const getRadioButton = wrapper => wrapper.find(RadioButton);
```

Comment:
is there a reason why we shouldn't return that `RadioButton` directly in the `render` method?

Reply:
Doing so would prevent us from calling methods on the wrapper itself. I believe `wrapper.update()` has to be called at the top level. There are no calls to `wrapper.update()` at the moment, but changing that would make the tests hard to extend in the future.

Reply:
good point